CycloneDX/Program.cs

```diff
@@ -305,10 +308,12 @@ public static async Task<int> Main(string[] args)
             {
                 return (int)ExitCode.GitHubLicenseResolutionFailed;
             }
+
+            var directPackageDependencies = packages.Where(p => p.IsDirectReference).Select(p => packageToComponent[p].BomRef).ToHashSet();
             // now we loop through all the dependencies to check which are direct
             foreach (var dep in dependencies)
             {
-                if (!transitiveDepencies.Contains(dep.Ref))
+                if (directPackageDependencies.Contains((dep.Ref)) || !transitiveDepencies.Contains(dep.Ref))
```

Comment:
Seems that I should have probably removed the old condition instead of combining it. 
This causes transitive framework dependencies to still be considered top level dependencies.

Reply:
Updated to prioritize directPackageDependencies when filled over list transitive dependencies.